partcad/src/partcad/adhoc/generate.py

```
@@ -0,0 +1,87 @@
+
+            current_dir = Path.cwd()
+            target_path = current_dir / Path(part.path).name
+            part.path = str(target_path)
+
+            asyncio.run(get_shape_async(part, ctx))
+
+            if part.errors:
+                raise RuntimeError(f"Failed to load the part: {part.errors}")
+
+    except Exception as e:
+        raise RuntimeError(f"Failed to generate: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)